### PR TITLE
Increased Court and DK jump logic flexibility

### PR DESCRIPTION
**Pull request by jaybuidl**
## PR-Codex overview
This PR focuses on refining the initialization parameters for various `DisputeKit` contracts by removing unnecessary arguments, enhancing the optimization settings, and implementing new logic for dispute kit transitions. It also adds tests for the new functionality.

### Detailed summary
- Removed `jumpDisputeKitID` argument from `initialize` functions in several `DisputeKit` contracts.
- Updated optimizer `runs` from 800 to 1000 in `hardhat.config.ts`.
- Simplified deployment arguments for `DisputeKitClassic`, `DisputeKitShutter`, and `DisputeKitGated`.
- Introduced `NextRoundSettings` struct in `DisputeKitClassicBase` for managing round settings.
- Updated `getNextRoundSettings` method to accommodate new logic for determining court and dispute kit transitions.
- Added tests for new functionality in `KlerosCore_AppealsTest`, including scenarios for court jumps and dispute kit changes.
- Ensured compatibility checks are enforced during dispute transitions.
- Implemented fallbacks for invalid jump settings and custom voting scenarios.

> The following files were skipped due to too many changes: `contracts/test/foundry/KlerosCore_Appeals.t.sol`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

* **New Features**
  * Public getter now returns next court, next dispute‑kit and resulting vote count; draw also returns originating subcourt ID.

* **Refactor**
  * Appeal/jump flow consolidated around compatibility‑based next‑round settings.
  * Per‑court NextRoundSettings added, configurable and emitting events; legacy jump helpers removed.

* **Chore**
  * Initializer signatures simplified across dispute‑kit variants; deployments/tests updated to match.
  * Solidity optimizer runs increased.

* **Tests**
  * Expanded appeal/jump test coverage, edge cases, and cost/funding scenarios.

### CODE REVIEW

File: contracts/src/arbitration/KlerosCore.sol [hunk 1]
```diff
@@ -1080,31 +1080,26 @@ contract KlerosCore is IArbitratorV2, Initializable, UUPSProxiable {
...
-            cost = court.feeForJuror * nbVotesAfterAppeal;
+        if (newCourtID == dispute.courtID) {
+            // No court jump
+            return court.feeForJuror * nbVotesAfterAppeal;
+        }
+        if (dispute.courtID != GENERAL_COURT && newCourtID != FORKING_COURT) {
+            // Court jump but not to the Forking court
+            return courts[newCourtID].feeForJuror * nbVotesAfterAppeal;
         }
+        return NON_PAYABLE_AMOUNT; // Jumping to the Forking Court is not supported yet.
```
[the remark this answers is not in the document]
Comment (jaybuidl) [reply contracts/src/arbitration/KlerosCore.sol]: It is implied that the General Court is the only sub-court of the Forking court so it's fine.
